server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -168,8 +173,12 @@ public void run() {
                     handleProducing(null, System.currentTimeMillis() / 1000L, KafkaRecordKind.NORMAL);
                 }
             }
-        } catch (Exception e) {
-            logger.error("Error occurred while processing call graphs", e);
+
+        } catch (WakeupException e) {
+            if (!closed.get()) throw e;
```

Comment:
A comment would help here to understand in which cases the exception does not need to be re-thrown. What is the semantic of `!closed.get()`?

Reply:
Addressed in bdb0474.

Reply:
Please rephrase: "WakeUpException is used to wake up waiting Kafka clients. If the connection is already closed, we can ignore the exception, otherwise, we re-throw it."

Reply:
(see comments below)